src/arkode/arkode_sunstepper.c

```diff
@@ -0,0 +1,164 @@
+
+static SUNErrCode arkSUNStepperFullRhs(SUNStepper stepper, sunrealtype t,
+                                       N_Vector y, N_Vector f)
+{
+  SUNFunctionBegin(stepper->sunctx);
+  /* extract the ARKODE memory struct */
+  void* arkode_mem;
+  SUNCheckCall(SUNStepper_GetContent(stepper, &arkode_mem));
+  ARKodeMem ark_mem = (ARKodeMem)arkode_mem;
+
+  stepper->last_flag = ark_mem->step_fullrhs(ark_mem, t, y, f, ARK_FULLRHS_OTHER);
+  if (stepper->last_flag != ARK_SUCCESS) { return SUN_ERR_OP_FAIL; }
+
+  return SUN_SUCCESS;
+}
+
+/*------------------------------------------------------------------------------
+  Implementation of SUNStepperResetFn to reset the inner (fast) stepper
+  state.
+  ----------------------------------------------------------------------------*/
+
+static SUNErrCode arkSUNStepperReset(SUNStepper stepper, sunrealtype tR,
+                                     N_Vector yR)
+{
+  SUNFunctionBegin(stepper->sunctx);
+  /* extract the ARKODE memory struct */
+  void* arkode_mem;
+  SUNCheckCall(SUNStepper_GetContent(stepper, &arkode_mem));
+
+  stepper->last_flag = ARKodeReset(arkode_mem, tR, yR);
+  if (stepper->last_flag != ARK_SUCCESS) { return SUN_ERR_OP_FAIL; }
+
+  return SUN_SUCCESS;
+}
+
+/*------------------------------------------------------------------------------
+  Implementation of SUNStepperStopTimeFn to set the tstop time
+  ----------------------------------------------------------------------------*/
+
+static SUNErrCode arkSUNStepperSetStopTime(SUNStepper stepper, sunrealtype tstop)
+{
+  SUNFunctionBegin(stepper->sunctx);
+  /* extract the ARKODE memory struct */
+  void* arkode_mem;
+  SUNCheckCall(SUNStepper_GetContent(stepper, &arkode_mem));
+
+  stepper->last_flag = ARKodeSetStopTime(arkode_mem, tstop);
+  if (stepper->last_flag != ARK_SUCCESS) { return SUN_ERR_OP_FAIL; }
+
+  return SUN_SUCCESS;
+}
+
+static SUNErrCode arkSUNStepperSetForcing(SUNStepper stepper, sunrealtype tshift,
+                                          sunrealtype tscale, N_Vector* forcing,
+                                          int nforcing)
+{
+  SUNFunctionBegin(stepper->sunctx);
+  /* extract the ARKODE memory struct */
+  void* arkode_mem;
+  SUNCheckCall(SUNStepper_GetContent(stepper, &arkode_mem));
+  ARKodeMem ark_mem = (ARKodeMem)arkode_mem;
+
+  stepper->last_flag = ark_mem->step_setforcing(ark_mem, tshift, tscale,
+                                                forcing, nforcing);
+  if (stepper->last_flag != ARK_SUCCESS) { return SUN_ERR_OP_FAIL; }
+
+  return SUN_SUCCESS;
+}
+
+int ARKodeCreateSUNStepper(void* arkode_mem, SUNStepper* stepper)
+{
+  /* unpack ark_mem */
+  if (arkode_mem == NULL)
+  {
+    arkProcessError(NULL, ARK_MEM_NULL, __LINE__, __func__, __FILE__,
+                    MSG_ARK_NO_MEM);
+    return (ARK_MEM_NULL);
+  }
+  ARKodeMem ark_mem = (ARKodeMem)arkode_mem;
+
+  SUNErrCode err = SUNStepper_Create(ark_mem->sunctx, stepper);
+  if (err != SUN_SUCCESS) { return ARK_SUNSTEPPER_ERR; }
```

Comment:
I don't feel strongly on this question either way.  While it is an ARKODE-provided routine (and so `arkProcessError` makes sense), it is a utility routine provided for `SUNStepper` users, so I could equivalently see it returning a `SUNErrCode` instead of an `int`, and "feeling" more similar to the SUNStepper routines.